lib/ssl/src/inet_tls_dist.erl

```diff
@@ -591,6 +591,7 @@ do_setup_connect(Driver, Kernel, Node, Address, Ip, TcpPort, Version, Type, MyNo
         net_kernel:connecttime()) of
     {ok, #sslsocket{pid = [_, DistCtrl| _]} = SslSocket} ->
             _ = monitor_pid(DistCtrl),
+            Kernel ! {post_setup, MyNode, ?MODULE, DistCtrl},
```

Comment:
Sending it to net_kernel in a message is not great idea because by the time when we start extracting the key.log, the connection can already be gone. It's a race basically. I think we should log it here synchronously instead.
I understand that we don't have "Mod" here so I guess we have two options:
1. Pass the name of the module as kernel application env (see application:get_env/put_env)
2. Hardcode the name of the module (check erlang:function_exported before calling)

Reply:
I have done option 1.